src/renderer/qss/js/qssFocusManager.js

```diff
@@ -0,0 +1,343 @@
+        var focusGroups = [],
+            styles = that.options.styles,
+            qssButtons = container.find("." + styles.button + ":visible"),
+            currentFocusGroup = [];
+
+        fluid.each(qssButtons, function (qssButton) {
+            if (jQuery(qssButton).hasClass(styles.smallButton)) {
+                currentFocusGroup.push(qssButton);
+                // Mark the group as complete if the max number of elements has been reached.
+                if (currentFocusGroup.length >= that.options.maxElementsPerFocusGroup) {
+                    focusGroups.push(currentFocusGroup);
+                    currentFocusGroup = [];
+                }
+            } else {
+                // A "big" button can only be added to a group where it is the only element.
+                // So complete any partial group that may exist before that.
+                if (currentFocusGroup.length > 0) {
+                    focusGroups.push(currentFocusGroup);
+                    currentFocusGroup = [];
+                }
+                focusGroups.push([qssButton]);
+            }
+        });
+
+        return focusGroups;
+    };
+
+    /**
+     * Retrieves information about the focusable elements in the QSS.
+     * @param {Component} that - The `gpii.qss.qssFocusManager` instance.
+     * @param {jQuery} container - The jQuery element representing the container of the component.
+     * @return {FocusGroupsInfo} - A `FocusGroupsInfo` object with the requested information.
+     */
+    gpii.qss.qssFocusManager.getFocusGroupsInfo = function (that, container) {
+        var focusGroups = gpii.qss.qssFocusManager.getFocusGroups(that, container),
+            focusedElement = container.find("." + that.options.styles.focused)[0];
+
+        var focusIndex = -1,
+            focusGroupIndex = fluid.find(focusGroups, function (focusGroup, index) {
+                focusIndex = jQuery.inArray(focusedElement, focusGroup);
+                if (focusIndex > -1) {
+                    return index;
+                }
+            }, -1);
+
+        return {
+            focusGroups: focusGroups,
+            focusGroupIndex: focusGroupIndex,
+            focusIndex: focusIndex
+        };
+    };
+
+    /**
+     * Focuses the first available button which conforms to all of the following conditions:
+     * 1. The button is focusable.
+     * 2. The button is in the same focus group as the currently focused button.
+     * 3. The button is the first button before or after the currently focused button (depending
+     * on the `backwards` argument) which conforms to the two conditions above.
+     * @param {Component} that - The `gpii.qss.qssFocusManager` instance.
+     * @param {Boolean} backwards - If `true` the scanning direction will be from bottom to top.
+     * Otherwise, it will be from top to bottom.
+     */
+    gpii.qss.qssFocusManager.focusNearestVertically = function (that, backwards) {
```

Comment:
It is these which should be the exposed operations on the component, rather than being just used in the impl - people bringing alternative interactions/a11y technologies to the component will want to talk in terms of the focus operation rather than the name of the keypress.